packages/server/src/core/bank_rule/iterate_courses.rs

```diff
@@ -45,25 +23,53 @@ impl<'a> BankRuleHandler<'a> {
                         course_status.course.id.clone(),
                     );
                 } else {
+                    #[inline(always)]
```

Comment:
Do we need this tag? The compiler should decide on its own if better to inline the function or not, shouldn't it?

Reply:
This should definitely be inlined as it was meant as a private function with the sole purpose of saving duplication of code in 2 specific places. Furthermore, inlining this brings us closer to the old behavior (macro) by having the advantage of not paying the extra cost for the function invocation _and_ also using a function over a macro.